bothan-api/server/src/monitoring/client.rs

```diff
@@ -0,0 +1,68 @@
+use rand::rngs::OsRng;
+use rand::RngCore;
+use reqwest::header::HeaderMap;
+use reqwest::Response;
+use serde::Serialize;
+
+use bothan_core::store::ActiveSignalIDs;
+
+use crate::monitoring::error::PostError;
+use crate::monitoring::signer::Signer;
+use crate::monitoring::types::{BothanInfo, Entry, Topic};
+use crate::proto::price::Price;
+use crate::VERSION;
+
+pub struct Client {
+    url: String,
+    client: reqwest::Client,
+    signer: Signer,
+}
+
+impl Client {
+    pub fn new<T: Into<String>>(url: T, signer: Signer) -> Self {
+        Self {
+            url: url.into(),
+            client: reqwest::Client::new(),
+            signer,
+        }
+    }
+
+    pub async fn post_price(&self, prices: Vec<Price>) -> Result<Response, PostError> {
+        self.post(Topic::Price, prices).await
+    }
+
+    pub async fn post_heartbeat(
+        &self,
+        active_signal_ids: ActiveSignalIDs,
+    ) -> Result<Response, PostError> {
+        let bothan_info = BothanInfo::new(
+            VERSION.to_string(),
```

Comment:
Just a question, is version alone enough to derive how many sources this validator support?

Reply:
no, this is just the current bothan version that the validator is running